wire-runtime/build.gradle

```diff
@@ -1,12 +1,98 @@
-apply plugin: 'org.jetbrains.kotlin.jvm'
-
-dependencies {
-  api deps.okio
-  api deps.kotlin.stdlib.jdk6
-  compileOnly deps.android
-  compileOnly deps.jsr305
-  testImplementation deps.junit
-  testImplementation deps.assertj
+apply plugin: 'org.jetbrains.kotlin.multiplatform'
+
+kotlin {
+  jvm()
+  js() {
+    configure([compilations.main, compilations.test]) {
+      tasks.getByName(compileKotlinTaskName).kotlinOptions {
+        moduleKind = "commonjs"
+      }
+    }
+  }
+  iosX64() {
+    binaries {
+      framework()
+    }
+  }
+  iosArm64() {
+    binaries {
+      framework()
+    }
+  }
+  // Required to generate tests tasks: https://youtrack.jetbrains.com/issue/KT-26547
+  linuxX64()
+  macosX64()
+  mingwX64('winX64')
+  sourceSets {
+    commonMain {
+      dependencies {
+        api deps.kotlin.stdlib.common
+      }
+    }
+    commonTest {
+      dependencies {
+        implementation deps.kotlin.test.common
+        implementation deps.kotlin.test.annotations
+      }
+    }
+    jvmMain {
+      dependencies {
+        api deps.kotlin.stdlib.jdk6
+        api deps.okio
+        compileOnly deps.android
+      }
+    }
+    jvmTest {
+      dependencies {
+        implementation deps.assertj
+        implementation deps.kotlin.test.junit
+      }
+    }
+    test {
+      // Not required, but helps IntelliJ configure inter-project dependencies:
+      // https://youtrack.jetbrains.com/issue/KT-31179
+      dependsOn jvmTest
+    }
+    jsMain {
+      dependencies {
+        api deps.kotlin.stdlib.js
+      }
+    }
+    jsTest {
+      dependencies {
+        implementation deps.kotlin.test.js
+      }
+    }
+    nativeMain {
+      dependsOn commonMain
```

Comment:
These are needed?

Reply:
Copied from Okio, I believe @bnorm added those to fix IDE issues

Reply:
https://kotlinlang.org/docs/reference/building-mpp-with-gradle.html#connecting-source-sets

> Custom source sets created in addition to the default ones should be explicitly included into the dependencies hierarchy to be able to use declarations from other source sets and, most importantly, to take part in compilations. Most often, they need a dependsOn(commonMain) or dependsOn(commonTest) statement, and some of the default platform-specific source sets should depend on the custom ones, directly or indirectly:

This is a weird intersection where IntelliJ is unhappy with all the code in `nativeMain` without the `dependsOn` because it is correctly interpreting the rules but Gradle is happy without it because it never actually compiles `nativeMain`, it compiles `linuxX64Main` which is a default source set so it automatically includes `commonMain` and we're adding `nativeMain` to that so everything resolves.

Is this an IntelliJ bug because it displays code red when it actually compiles? Is this a compiler bug because `nativeMain` doesn't actually depend on `commonMain`? Is it not a bug at all because the documentation clear states that you need this `dependsOn`?